conveyer/queue/queue_interface.go

```diff
@@ -0,0 +1,82 @@
+/*
+ *    Copyright 2019 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package queue
+
+// SyncDone is callback for biggestQueueSignal
+type SyncDone interface {
+	done()
+}
+
+// OutputElement represent one element returned from queue
+type OutputElement struct {
+	data     interface{}
+	itemType uint32
+}
+
+type queueItem struct {
+	itemType           uint32
+	biggestQueueSignal uint32
+	index              uint
+	payload            interface{}
+	next               *queueItem
+}
+
+func (qi *queueItem) isSignal() bool {
+	return qi.itemType != 0
+}
+
+func (qi *queueItem) hasSignal() bool {
+	return qi.biggestQueueSignal != 0
+}
+
+var emptyQueueItem queueItem
```

Comment:
can we just initialize emptyQueueItem here without init() function?

Reply:
Yes, we can.
But what for?